components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,49 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuturedCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, v.len()));
+    }
+    unsafe fn write_usize(&self, mut u: &usize) {
+        write_length(self.w, u.to_be());
+    }
+}
+
+struct StructuredCloneReader {
+    r: *mut JSStructuredCloneReader,
+}
+
+impl StructuredCloneReader {
+    unsafe fn read_bytes(&self) {
+        assert!(JS_ReadBytes(self.r, vec![0u8; self.read_usize()].as_mut_ptr() as *mut raw::c_void, self.read_usize()));
```

Comment:
This reads two separate usize values, rather than reading one and then using that value as the vector length as well as number of bytes to read. Please use a local variable instead.